plugins/DataLoadRLog/dataload_rlog.cpp

```diff
@@ -0,0 +1,164 @@
+#include "dataload_rlog.hpp"
+#include "rlog_parser.hpp"
+#include <iostream>
+
+DataLoadRLog::DataLoadRLog(){
+  _extensions.push_back("bz2"); 
+}
+
+DataLoadRLog::~DataLoadRLog(){
+}
+
+const std::vector<const char*>& DataLoadRLog::compatibleFileExtensions() const{
+  return _extensions;
+}
+
+bool DataLoadRLog::readDataFromFile(FileLoadInfo* fileload_info, PlotDataMapRef& plot_data){
+  
+  Events events;
+
+  QProgressDialog progress_dialog;
+  progress_dialog.setLabelText("Loading... please wait");
+  progress_dialog.setWindowModality(Qt::ApplicationModal);
+  progress_dialog.setRange(0, 3);
+  progress_dialog.show();
+  
+  auto fn = fileload_info->filename;
+  qDebug() << "Loading: " << fn;
+
+  // Load file:
+  bz_stream bStream;
```

Comment:
abstract the decompression out into its own function

Reply:
you can also simplify the decompression with `BZ2_bzReadOpen `  https://sourceware.org/bzip2/manual/manual.html#bzreadopen